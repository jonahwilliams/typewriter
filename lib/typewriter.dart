// Copyright (c) 2017, Jonah Williams. All rights reserved. Use of this source code
// is governed by a BSD-style license that can be found in the LICENSE file.

/// Support for doing something awesome.
///
/// More dartdocs go here.
library typewriter;

export 'analysis/analysis.dart';
export 'annotations/annotations.dart';
export 'generators/generator.dart';

// TODO: Export any libraries intended for clients of this package.
