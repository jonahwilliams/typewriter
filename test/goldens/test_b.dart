library typewriter.testing.test_b;

import 'package:typewriter/typewriter.dart';

@Json()
class Person {
  List<String> items;
  double age;
  int coins;
  Dog dog;
  Cat cat;
}

@Json()
class Dog {
  String name;

}

@Json()
class Cat {
  String name;

  @JsonKey('birth_day')
  DateTime birthday;
}
