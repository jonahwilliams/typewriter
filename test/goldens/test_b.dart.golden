part of typewriter.testing.test_b;

const jsonCodec = const JsonCodec();

class PersonDecoder extends Converter<Object, Person> {
  const PersonDecoder();

  @override
  Person convert(Object raw) {
    var input = raw as Map<String, dynamic>;
    var output = new Person();
    output.items = input["items"].map((x) => x).toList();
    output.age = input["age"];
    output.coins = input["coins"];
    output.dog = const DogDecoder().convert(input["dog"]);
    output.cat = const CatDecoder().convert(input["cat"]);
    return output;
  }
}

class PersonEncoder extends Converter<Person, Object> {
  const PersonEncoder();

  @override
  Object convert(Person input) {
    var output = <String, dynamic>{};
    output["items"] = input.items.map((x) => x).toList();
    output["age"] = input.age;
    output["coins"] = input.coins;
    output["dog"] = const DogEncoder().convert(input.dog);
    output["cat"] = const CatEncoder().convert(input.cat);
    return output;
  }
}

class PersonCodec extends Codec<Person, Object> {
  const PersonCodec();

  @override
  Converter<Object, Person> get decoder => const PersonDecoder();

  @override
  Converter<Person, Object> get encoder => const PersonEncoder();
}

class DogDecoder extends Converter<Object, Dog> {
  const DogDecoder();

  @override
  Dog convert(Object raw) {
    var input = raw as Map<String, dynamic>;
    var output = new Dog();
    output.name = input["name"];
    return output;
  }
}

class DogEncoder extends Converter<Dog, Object> {
  const DogEncoder();

  @override
  Object convert(Dog input) {
    var output = <String, dynamic>{};
    output["name"] = input.name;
    return output;
  }
}

class DogCodec extends Codec<Dog, Object> {
  const DogCodec();

  @override
  Converter<Object, Dog> get decoder => const DogDecoder();

  @override
  Converter<Dog, Object> get encoder => const DogEncoder();
}

class CatDecoder extends Converter<Object, Cat> {
  const CatDecoder();

  @override
  Cat convert(Object raw) {
    var input = raw as Map<String, dynamic>;
    var output = new Cat();
    output.name = input["name"];
    output.birthday = DateTime.parse(input["birth_day"]);
    return output;
  }
}

class CatEncoder extends Converter<Cat, Object> {
  const CatEncoder();

  @override
  Object convert(Cat input) {
    var output = <String, dynamic>{};
    output["name"] = input.name;
    output["birth_day"] = input.birthday.toIso8601String();
    return output;
  }
}

class CatCodec extends Codec<Cat, Object> {
  const CatCodec();

  @override
  Converter<Object, Cat> get decoder => const CatDecoder();

  @override
  Converter<Cat, Object> get encoder => const CatEncoder();
}